R/lazyframe__lazy.R

```diff
@@ -1070,19 +1101,32 @@ LazyFrame_group_by = function(..., maintain_order = pl$options$maintain_order) {
 #' df1 = pl$LazyFrame(x = letters[1:3])
 #' df2 = pl$LazyFrame(y = 1:4)
 #' df1$join(other = df2, how = "cross")
+#'
+#' # use "validate" to ensure join keys are not duplicated
+#' df1 = pl$LazyFrame(x = letters[1:5], y = 1:5)
+#' df2 = pl$LazyFrame(x = c("a", letters[1:4]), y2 = 6:10)
+#'
+#' # this throws an error because there are two keys in df2 that match the key
+#' # in df1
+#' tryCatch(
+#'   df1$join(df2, on = "x", validate = "1:1")$collect(),
+#'   error = function(e) print(e)
+#' )
 LazyFrame_join = function(
-    other, # : LazyFrame or DataFrame,
-    left_on = NULL, # : str | pli.RPolarsExpr | Sequence[str | pli.RPolarsExpr] | None = None,
-    right_on = NULL, # : str | pli.RPolarsExpr | Sequence[str | pli.RPolarsExpr] | None = None,
-    on = NULL, # : str | pli.RPolarsExpr | Sequence[str | pli.RPolarsExpr] | None = None,
-    how = c("inner", "left", "outer", "semi", "anti", "cross"),
+    other,
+    on = NULL,
+    how = c("inner", "left", "outer", "semi", "anti", "cross", "outer_coalesce"),
+    left_on = NULL,
+    right_on = NULL,
     suffix = "_right",
+    validate = "m:m",
+    join_nulls = FALSE,
     allow_parallel = TRUE,
     force_parallel = FALSE) {
   uw = \(res) unwrap(res, "in $join():")
 
-  if (inherits(other, "RPolarsDataFrame")) {
-    other = other$lazy()
+  if (!inherits(other, "RPolarsLazyFrame")) {
+    Err_plain("`other` must be a LazyFrame.") |> uw()
```

Comment:
- Could you please note this change in the NEWS?
- Could you use `is_polars_lf` instead of `inherits`?